cmd/bisync/cmd.go

```
@@ -0,0 +1,224 @@
+	flags.BoolVarP(cmdFlags, &Opt.Force, "force", "", Opt.Force, "Bypass --max-delete safety check and run the sync. Consider using with --verbose")
+	flags.FVarP(cmdFlags, &Opt.CheckSync, "check-sync", "", "Controls comparison of final listings: true|false|only (default: true)")
+	flags.BoolVarP(cmdFlags, &Opt.RemoveEmptyDirs, "remove-empty-dirs", "", Opt.RemoveEmptyDirs, "Remove empty directories at the final cleanup step.")
+	flags.StringVarP(cmdFlags, &Opt.FiltersFile, "filters-file", "", Opt.FiltersFile, "Read filtering patterns from a file")
+	flags.StringVarP(cmdFlags, &Opt.Workdir, "workdir", "", Opt.Workdir, makeHelp("Use custom working dir - useful for testing. (default: {WORKDIR})"))
+	flags.BoolVarP(cmdFlags, &tzLocal, "localtime", "", tzLocal, "Use local time in listings (default: UTC)")
+	flags.BoolVarP(cmdFlags, &Opt.NoCleanup, "no-cleanup", "", Opt.NoCleanup, "Retain working files (useful for troubleshooting and testing).")
+}
+
+// bisync command definition
+var commandDefinition = &cobra.Command{
+	Use:   "bisync remote1:path1 remote2:path2",
+	Short: shortHelp,
+	Long:  longHelp,
+	RunE: func(command *cobra.Command, args []string) error {
+		cmd.CheckArgs(2, 2, command, args)
+		fs1, file1, fs2, file2 := cmd.NewFsSrcDstFiles(args)
+		if file1 != "" || file2 != "" {
+			return errors.New("paths must be existing directories")
+		}
+
+		ctx := context.Background()
+		opt := Opt
+		opt.applyContext(ctx)
+
+		if tzLocal {
+			TZ = time.Local
+		}
+
+		commonHashes := fs1.Hashes().Overlap(fs2.Hashes())
+		isDropbox1 := strings.HasPrefix(fs1.String(), "Dropbox")
+		isDropbox2 := strings.HasPrefix(fs2.String(), "Dropbox")
+		if commonHashes == hash.Set(0) && (isDropbox1 || isDropbox2) {
+			ci := fs.GetConfig(ctx)
+			if !ci.DryRun && !ci.RefreshTimes {
+				fs.Debugf(nil, "Using flag --refresh-times is recommended")
+			}
+		}
+
+		// FIXME Why runtime errors make rclone print usage?
```

@ncw 
Why runtime errors make cobra print usage?
How can I disable that?

---

I guess that the cobra framework in rclone expects me to use only `cmdDefinition.Run(myFuncWithoutErrorReturn)`,
never return errors and use the `md.Run(myFailingFunc)` high-level retries inside.
But what if I want `cmdDefinition.RunE(myFuncWithError)` so Bisync can just fail fast?
I'm still confused why **any** failure is considered a CLI syntax error calling for help?
Please advise.

---

@ncw 
https://github.com/rclone/rclone/pull/5164/commits/d94b77b3dd4d2e47be2eaee3ff8cf4b558eba2a6
Does this fix look right?

---

@ncw 
Does the fix https://github.com/rclone/rclone/pull/5164/commits/d94b77b3dd4d2e47be2eaee3ff8cf4b558eba2a6 look right for you?